agent/consul/rpc.go

```diff
@@ -542,18 +543,38 @@ func (s *Server) blockingQuery(queryOpts structs.QueryOptionsCompat, queryMeta s
 	timeout = time.NewTimer(queryTimeout)
 	defer timeout.Stop()
 
+	// REVIEW: The spec for this placed the atomic inc for queries_blocking inside the RUN_QUERY statement
+	// however, there are a combination of factors that I believe should mean we should place it before.
+	// - `goto RUN_QUERY` on L528 above means that we'd instrument a non-blocking query as blocking for its duration.
+	// - using defer dec on L554 would mean that we inc the counter multiple times as the blocking query waits, while
+	//   we only dec on the return of blockingQuery(). We'd need swap out defer for a dec on retry on L616 as well as
+	// 	 on the return at L621
+	// -> There I think it's simpler if we don't instrument the RUN_QUERY statement and continue to use the
+    // 	  end of s.blockingQuery()'s setup phase as the point of measurement.
+
+	// atomic inc our gauge of blockingQueries
+	atomic.AddUint64(&s.queriesBlocking, 1)
+	// atomic dec when we return from blockingQuery()
+	defer atomic.AddUint64(&s.queriesBlocking, -1)
```

Comment:
The docs for `AddUint64` are saying:

>  In particular, to decrement x, do AddUint64(&x, ^uint64(0)). 

Wonder if we should do that. I am actually not sure.

Reply:
Ran it in a scratch and it extremely panics. Not very intuitive but I can see why -- thank you for double checking this

Reply:
```
func main() {
	var counter uint64
	atomic.AddUint64(&counter, 1)
	atomic.AddUint64(&counter, ^uint64(0))
	atomic.AddUint64(&counter, 1)
	atomic.AddUint64(&counter, 1)
	atomic.AddUint64(&counter, ^uint64(0))

	ok := 1 == atomic.LoadUint64(&counter)

	fmt.Printf("Is counter 1? %v", ok) // => Is counter 1? true
}
```